src/xrpld/app/tx/detail/MPTokenIssuanceCreate.cpp

```diff
@@ -0,0 +1,142 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2024 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <xrpld/app/tx/detail/MPTokenIssuanceCreate.h>
+#include <xrpld/ledger/View.h>
+#include <xrpl/protocol/Feature.h>
+#include <xrpl/protocol/TxFlags.h>
+#include <xrpl/protocol/st.h>
+
+namespace ripple {
+
+NotTEC
+MPTokenIssuanceCreate::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureMPTokensV1))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
```

Comment:
These branches, here and elsewhere, can all be simplified.

```suggestion
    if (auto const ret = preflight1(ctx))
```

Reply:
I think this is fine. Every other transactions has this.

Reply:
"Other transactions have this" is not a good reason. This simplification works because a `TER` evaluates to `true` if it is _not_ success, just like a Linux error code.

Reply:
You know, I've gotten so used to seeing the symbolic `isTesSuccess`, that I've never really thought about the fact that `tesSUCCESS == 0`. But, yeah, this optimization / simplification should work.

The question is what does it gain us?
On one hand, if I see `isTesSuccess` (or any of the other `is*` test functions), I don't have to think about the underlying enum values. Additionally, `isTesSuccess` is defined as
```
inline bool
isTesSuccess(TER x)
{
    return ((x) == tesSUCCESS);
}
```
Since that's `inline`, when compiled it should just boil down to `x == 0`. As an optimization, this probably doesn't gain much.

On the other hand, as a simplification, it does read a lot nicer, and fits the pattern we use everywhere else the result can be evaluated as a `bool`. I think I personally would adjust pretty quickly to seeing it this way, so I still wouldn't have to think about the underlying values.

My vote is to remove the call to `isTesSuccess`, and optionally add a comment anywhere it's removed to remind readers that this works.
```suggestion
    // tesSUCCESS = 0, so any other return will be returned.
    if (auto const ret = preflight1(ctx))
```

Reply:
Another idea, similar to my first suggestion:
```suggestion
    // tesSUCCESS == 0. Any other value indicates an error.
    if (auto const err = preflight1(ctx))
```
(then `return err;`)
Using the name `err` instantly communicates "This function will return an error if it doesn't succeed."